presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestMultipleDistinctAggregationToMarkDistinct.java

```diff
@@ -0,0 +1,121 @@
+                                        p.symbol("input2")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testSingleDistinct()
+    {
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input1)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(
+                                        p.symbol("input1"),
+                                        p.symbol("input2")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testMultipleAggregations()
+    {
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input)"), ImmutableList.of(BIGINT))
+                        .addAggregation(p.symbol("output2"), expression("sum(DISTINCT input)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(p.symbol("input")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDistinctWithFilter()
+    {
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input1) filter (where input2 > 0)"), ImmutableList.of(BIGINT))
+                        .addAggregation(p.symbol("output2"), expression("count(DISTINCT input2) filter (where input1 > 0)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(
+                                        p.symbol("input1"),
+                                        p.symbol("input2")))))
+                .doesNotFire();
+
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input1) filter (where input2 > 0)"), ImmutableList.of(BIGINT))
+                        .addAggregation(p.symbol("output2"), expression("count(DISTINCT input2)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(
+                                        p.symbol("input1"),
+                                        p.symbol("input2")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testMixedDistinctAndNonDistinct()
+    {
+//        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
```

Comment:
uncomment?

Reply:
I see your comment. Will you still work on that?

Reply:
I'm don't think I'll be able to get to that in the near future. I'll leave those tests for later. There's enough coverage with the end-to-end tests I added in this commit and to ensure the rule doesn't fire when it shouldn't.

Reply:
> There's enough coverage with the end-to-end tests I added in this commit and to ensure the rule doesn't fire when it shouldn't.

Right, but was it rewritten previously? Could not rewiring mixed dist aggregation have performance impact of some queries?

Do you consider this PR done?

Reply:
Not sure I understand your question. If it’s not rewritten, the query will fail because the operators cannot deal with aggregations that have a DISTINCT in them.

Yes, other than those missing tests, this should be done now.

Reply:
> If it’s not rewritten, the query will fail because the operators cannot deal with aggregations that have a DISTINCT in them.

What rewrites mixed aggregations with different distinct sets after your changes? e.g:
```count(distinct a), count(distinct b), count(c)```.

I'm probably missing some context here.

Reply:
Originally, this was all handled during initial SQL -> IR translation. Now, all that logic is in this rule. What I meant above is that if this rule doesn't catch something, it will just fail with a planning error. It won't affect performance.